Add instructions for nginx without ingress controllers

beyang opened the pull request.

keegancsmith commented on base/config-file.ConfigMap.yaml:
```diff
@@ -17,7 +17,9 @@ data:
-      "auth.providers": [{"type": "builtin", "allowSignup": false}]
+      "auth.providers": [{"type": "builtin", "allowSignup": false}],
+
+      "licenseKey": "eyJzaWciOnsiRm9ybWF0Ijoic3NoLXJzYSIsIkJsb2IiOiJKWlRJam8xZzV3TlBrcDE3VUlGekVnN1o5a25BcUR1aUNsb1FpbHNXbU1GV0ZkZ2tueUhCMVNWYmJDWFdrT3FZaDB2SHk0bWFGYXd4VkpDdTMyNTU3cW96SmhUd1hLcWlYTU9BRzFvb0s3Y09NTlI4SEtNbmYzS2RFUkI0czR4YzM3RmRnODRIZWlBdHNHeWwwejVHNkFjOVB1RmNQM2kvUWQyeWdjdTdpdUFQRnJtNVpQQWJSa0dXaktmL1F4WDVScENFOHpWd2dKZmViclphSmZWUnozRHR0eUdBYThhczYzMlgzZk1NNlJ6c2F0NS8yT3BHRFFhbW ...
```
@beyang did you mean to add this licenseKey into this repo?

keegancsmith commented on base/frontend/sourcegraph-frontend.Service.yaml:
```diff
@@ -11,9 +11,8 @@ metadata:
   selector:
     app: sourcegraph-frontend
-  type: NodePort
+  type: ClusterIP
```
If people try apply this, won't it fail if the service is already exists (as a NodePort)?

keegancsmith replied:
I agree, it was a change I wanted to make but did not due to this issue. If so we need to document it in the migration guide. An alternative solution is to update the selector on this service so it matches nothing, and create a new service. I'm unsure which is preferred. Likely the former since an admin upgrading would likely want to be aware of this sort of change (and will need to be doing changes for ingress-nginx anyways).

nicksnyder commented:
What is the decision here? In the absence of a decision or merging this, whatever is in master is going to get tagged as 3.0.0.

beyang commented:
@ggilmore PTAL (addressed your comments from https://sourcegraph.slack.com/archives/D1B7C7WQN/p1548724204000200).

Re the config map: changes to the config map will not propagate to nginx without a restart, but that should be known to anyone who's running nginx (i.e., I think discussing this is beyond the scopes of what our docs should be)

beyang commented:
I'll own deploying to sourcegraph.com

beyang commented:
Merged manually in 5a3afdfa1968c610b9a121ee3ce9eeb08232007c
